src/OSPSuite.UI/Binders/CurveBinder.cs

```diff
@@ -309,35 +309,52 @@ private void refreshData()
          if (!dimensionsConsistentToAxisUnits())
             return;
 
+         var baseGrid = activeBaseGrid(Curve.xData, ActiveYData);
```

Comment:
Sure, if my understanding of the usage is not quite right, then an interpolation step should not be lost. I thought it looked like in this case of refreshData, there would always be a 1-1 relationship between basegrid and curve data.

If that's wrong, then I'll add an interpolation, but I'll still try to use the index whenever possible without having to look it up.

Reply:
But it should be possible to use the index and value of basegrid together to prevent using IndexOf.
In this loop in refreshData we can compare the index and value of the iterated basegrid to the curve basegrid. If the curve base grid value at a given base grid index is the same, then we can use the value directly. 

That's what GetValue does anyway. If it happens to find an index of a basegrid value, then return the value directly without interpolation.

In this latest change, refreshData checks this directly instead of making any calls to IndexOf.

Check it out.